src/Microsoft.AspNetCore.Server.IISIntegration/IISMiddleware.cs

```diff
@@ -57,7 +57,7 @@ public IISMiddleware(RequestDelegate next, ILoggerFactory loggerFactory, IOption
             if (!string.Equals(_pairingToken, httpContext.Request.Headers[MSAspNetCoreToken], StringComparison.Ordinal))
             {
                 _logger.LogTrace($"'{MSAspNetCoreToken}' does not match the expected pairing token '{_pairingToken}', skipping {nameof(IISMiddleware)}.");
```

Comment:
IMHO, you should consider updating the log level of this message to error (or critical, since it would indicate that the request wasn't handled by IIS, which may be terribad in production).